### PR TITLE
feat(polls): rework

### PR DESCRIPTION
**Pull request by Calinteodor**
1. Save poll.
2. Edit poll.
3. Save edited poll.
4. Send poll to other participants.

### CODE REVIEW

File: react/features/polls/components/AbstractPollCreate.tsx [hunk 1]
```diff
@@ -21,10 +24,10 @@ type InputProps = {
  **/
 export type AbstractProps = InputProps & {
     addAnswer: (index?: number) => void;
-    answers: Array<string>;
+    answers: Array<string> | any;
```
Comment (horymury) [inline react/features/polls/components/AbstractPollCreate.tsx:27]: would not change the type here

File: react/features/polls/components/web/PollItem.tsx [hunk 2]
```diff
@@ -14,9 +14,14 @@ interface IProps {
      */
     pollId: string;
 
+    /**
+     * Create mode control.
+     */
+    setCreateMode: Function;
```
Comment (horymury) [inline react/features/polls/components/web/PollItem.tsx:20]: setCreateMode: (mode: boolean) => void;

File: react/features/polls/components/web/PollItem.tsx [hunk 3]
```diff
@@ -26,7 +31,10 @@ const PollItem = React.forwardRef<HTMLDivElement, IProps>(({ pollId }: IProps, r
                     key = { pollId }
                     pollId = { pollId } />
                 : <PollAnswer
-                    pollId = { pollId } />
+                    pollId = { pollId }
+
+                    // @ts-ignore
```
Comment (horymury) [inline react/features/polls/components/web/PollItem.tsx:36]: could be removed once the Interface def is updated

**Review by horymury — DISMISSED** [review]

**Comment by horymury** [comment]
I would revisit the cases where ts-ignore is used and check whether we can have a way to get rid of it. Also , I would suggest avoiding using `any` for type defs except the cases in which there is no way of knowing it.

**Comment by Calinteodor** [comment]
> I would revisit the cases where ts-ignore is used and check whether we can have a way to get rid of it. Also , I would suggest avoiding using `any` for type defs except the cases in which there is no way of knowing it.

Thanks Horatiu! I will look into it!

File: react/features/polls/reducer.ts [hunk 4]
```diff
@@ -51,18 +53,34 @@ ReducerRegistry.register<IPollsState>('features/polls', (state = INITIAL_STATE,
...
+    case RECEIVE_POLL:
+    case SAVE_POLL: {
+        let newState;
+
+        if (action.edit) {
+            newState = {
+                ...state,
+                polls: {
+                    ...state.polls,
+                    [action]: {
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/polls/reducer.ts:67]: Ops.

**Review by horymury — CHANGES_REQUESTED** [review]

File: react/features/polls/components/AbstractPollAnswer.tsx [hunk 5]
```diff
@@ -40,23 +36,25 @@ export type AbstractProps = {
  * @param {React.AbstractComponent} Component - The concrete component.
  * @returns {React.AbstractComponent}
  */
-const AbstractPollAnswer = (Component: ComponentType<AbstractProps>) => (props: InputProps) => {
+const AbstractPollAnswer = (Component: ComponentType<AbstractProps>) => (props: AbstractProps) => {
```
Comment (horymury) [inline react/features/polls/components/AbstractPollAnswer.tsx:39]: please revert this - I think the new props (setCreateMode, ...) should go on InputProps type instead  - AbastractProps seem to be defined for the <Component component.

File: react/features/polls/components/native/PollItem.tsx [hunk 6]
```diff
@@ -27,8 +32,11 @@ const PollItem = ({ pollId }: IProps) => {
                 ? <PollResults
                     key = { pollId }
                     pollId = { pollId } />
+
+                // @ts-ignore
```
Comment (horymury) [inline react/features/polls/components/native/PollItem.tsx:36]: please remove once AbstractPollAnswer is fixed

File: react/features/polls/components/web/PollCreate.tsx [hunk 7]
```diff
@@ -1,13 +1,16 @@
...
 
 import { withPixelLineHeight } from '../../../base/styles/functions.web';
 import Button from '../../../base/ui/components/web/Button';
 import Input from '../../../base/ui/components/web/Input';
 import { BUTTON_TYPES } from '../../../base/ui/constants.web';
+import { editPoll } from '../../actions';
 import { ANSWERS_LIMIT, CHAR_LIMIT } from '../../constants';
 import AbstractPollCreate, { AbstractProps } from '../AbstractPollCreate';
 
+
```
Comment (horymury) [inline react/features/polls/components/web/PollCreate.tsx:13]: extra space

File: react/features/polls/components/web/PollItem.tsx [hunk 8]
```diff
@@ -25,8 +30,11 @@ const PollItem = React.forwardRef<HTMLDivElement, IProps>(({ pollId }: IProps, r
                 ? <PollResults
                     key = { pollId }
                     pollId = { pollId } />
+
+                // @ts-ignore
```
Comment (horymury) [inline react/features/polls/components/web/PollItem.tsx:34]: please remove once AbstractPollAnswer is fixed

**Review by horymury — CHANGES_REQUESTED** [review]

**Review by horymury — APPROVED** [review]

**Review by horymury — APPROVED** [review]

**Review by horymury — DISMISSED** [review]

**Review by horymury — APPROVED** [review]

**Pull request merged**